code/modules/power/supermatter/supermatter.dm

```diff
@@ -463,6 +463,13 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 		if(T)
 			var/obj/singularity/energy_ball/E = new(T)
 			E.energy = 200 //Gets us about 9 balls
+			explosion(get_turf(T), explosion_power * max(gasmix_power_ratio, 0.205) * 0.5 , explosion_power * max(gasmix_power_ratio, 0.205) + 2, explosion_power * max(gasmix_power_ratio, 0.205) + 4 , explosion_power * max(gasmix_power_ratio, 0.205) + 6, 1, 1)
+			qdel(src)
+			return
+	if(prob(power/50)) //near the threshold we have around 100% that the crystal will create the event
```

Comment:
Do an else if here instead of moving the end of the proc up. It'll have the same effect and be cleaner.